### ISSUE TITLE
Teuchos::ParameterList::set does not need overloads for both const char[] and char[]

### ISSUE THREAD

**Issue by mhoemmen** [trilinos/Trilinos#6089]
## Enhancement
@trilinos/teuchos

`Teuchos::ParameterList::set` should not need overloads for both `const char value[]` and `char value[]`; the former should suffice.  [This example](https://godbolt.org/z/hXjTcB) builds even with C++98:
```c++
#include <string>

class Foo {
public:
    void set(const std::string& value) {
        value_ = value;
    }
    void set(const char value[]) {
        set(std::string(value));
    }
    void set(char value[]) {
        set(std::string(value));
    }
    template<class T>
    void set(const T& /* value */) {}
private:
    std::string value_;
};

int main() {
    Foo f;
    f.set("foo");
    std::string s("bar");
    f.set(s);
    f.set(s.c_str());
    f.set(int(42));
    return 0;
}
```

**Comment by bartlettroscoe** [comment]
I agree.  No idea why the non-const `char value[]` overload would exist.

**Comment by mhoemmen** [comment]
I suspect "something didn't compile" on some old compiler and this fixed it.  When I first started, the compilers we had to support seemed less consistent about their ability to deduce template arguments of templated instance methods.

**Comment by bartlettroscoe** [comment]
Still valid

**Comment by mhoemmen** [comment]
@bartlettroscoe Trilinos' project management is no longer a thing I need to worry about, thankfully, but it would be nice if the autocloser applied a tag ("wontfix"? "automatically closed"?).

**Comment by bartlettroscoe** [comment]
> but it would be nice if the autocloser applied a tag ("wontfix"? "automatically closed"?).

@mhoemmen, that is a good suggestion.  I will pass that up the chain.
